src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,70 @@
-import React from 'react';
+import React, { useState, useEffect, FormEventHandler } from 'react';
+import classNames from 'classnames';
 import './FindMovie.scss';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieData } from '../../types/MovieData';
+import { ResponseError } from '../../types/ReponseError';
+import { MovieCard } from '../MovieCard';
+import { DEFAULT_IMG } from './FindMovie.constants';
+
+interface Props {
+  onMovieAdd: (movie: Movie) => void;
+}
+
+export const FindMovie: React.FC<Props> = React.memo(({ onMovieAdd }) => {
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [query, setQuery] = useState('');
+  const [error, setError] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (query) {
+      setError(null);
+    }
+  }, [query]);
+
+  const handleSubmit: FormEventHandler<HTMLFormElement> = (event) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    getMovie(query)
+      .then((data: MovieData | ResponseError) => {
+        if ('Error' in data) {
```

Comment:
```suggestion
      .then((data: MovieData | ResponseError) => {
        if ('Error' in data) {
        const {Title, Plot, imdbID} = data
```
you can destructure data if you use it more than two times